instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -225,8 +225,10 @@ async def wrapped_send(message):
                         if send_span.is_recording():
                             if message["type"] == "http.response.start":
                                 status_code = message["status"]
+                                set_status_code(span, status_code)
                                 set_status_code(send_span, status_code)
```

Comment:
Does `send` here mean sending a response back to the client?
Can the op represented by the outer span have multiple sends or receives?

Reply:
As far as I know, the callable will be called once per connection, which is one request for HTTP. That should mean only one `http.response.start` event. Happy to make a change if that is wrong.

Reply:
I believe multiple calls to `receive` and `send` can occur per HTTP call like from [this issue](https://github.com/Azure/azure-sdk-for-python/issues/18171#issuecomment-827759899). Should the parent span then reflect ANY error that occurs during the flow? If this is the case, we probably need the response code logic in the `receive` as well.

Reply:
Yes, there are multiple send events, but the [second one](https://asgi.readthedocs.io/en/latest/specs/www.html#response-body-send-event) is the response body, and that event does not have another status code.

Reply:
@owais 
As discussed in the SIG meeting on 05/13/2021, we were wondering why there is a parent span for the channel in the first place? Would it be possible to remove this?

Reply:
I'll do a deep dive into ASGI and get back to this. As far as I understand right now, I think the receive span should be the SERVER span and "send" span should be it's child or grandchild. Both of these spans should use links to establish a relation with the channel span. I need some time to dig into this and confirm if my understanding is correct.